### DISCUSSION TITLE
Federating PubPub. Community Has No Boundary

### DISCUSSION THREAD

**Discussion by aschrijver** [knowledgefutures/pubpub#1249] [Ideas]
Hi there,

## TL;DR

What are the thoughts of those involved with [PubPub](https://www.pubpub.org/) on adding standards-based federation support? Because:

> **Community Has No Boundary**

#### About me

Short intro.. I am [Arnold Schrijver](https://www.linkedin.com/in/arnoldschrijver/), facilitator of the [Humane Tech Community](https://humanetech.community). Our [forum](https://community.humanetech.com) resides under the domain of [The Center for Humane Technology](https://humanetech.com), but we operate independently (@metasj of [Underlay](https://underlay.org) is admin with me). I am also an advocate of FOSS and [The Fediverse](https://en.wikipedia.org/wiki/Fediverse). And I would like to discuss the latter, in light of pubpub.

#### Fediverse

Now you may not have heard of the Fediverse, this grassroots conglomeration of open standards-based FOSS projects that federate together. You may have heard of some of its popular applications - they feature regularly on Hacker News nowadays - like [PeerTube](https://joinpeertube.org/) (federated/P2P YouTube alternative), [PixelFed](https://pixelfed.org) (Insta alternative) and most prominently [Mastodon](https://joinmastodon.org/) (Twitter alternative).

The open standards on which federation takes place are [W3C ActivityPub](https://www.w3.org/TR/activitypub/) and [ActivityStreams](https://www.w3.org/TR/activitystreams-vocabulary/). I co-maintain an [application watchlist](https://git.feneas.org/feneas/fediverse/-/wikis/watchlist-for-activitypub-apps) and one for [developer tools](https://git.feneas.org/feneas/fediverse/-/wikis/Watchlist-for-ActivityPub-developer-resources).

From a Humane Technology standpoint I consider the Fediverse as _the_ opportunity to do social media right this time! What is interesting is that, starting from small grassroots beginnings, convergence and coordinated efforts are emerging and great innovations are being worked on. In that light I want to mention [The Spritely Project](https://spritely.org) by AP specification co-author @cwebber and [openEngiadina](https://openengiadina.net/) by @pukkamustard and also complementary aspects to @timbl [Solid Project](https://forum.solidproject.org/t/discussion-solid-vs-activitypub/2685). Currently a federation protocol, in future we may move to a hybrid P2P/federated model.

Activities are steered from the [W3C SocialCG](https://www.w3.org/community/SocialCG/) (co-chaired by @rhiaro of the [W3C Technical Architecture Group](https://tag.w3.org/)) and [SocialHub](https://socialhub.activitypub.rocks/) forum.

#### ActivityPub-based Federation

Because Mastodon - a microblogging application - is so popular, the Fediverse is often thought of as 'just federated microblogging', and "being part of the Fediverse" means adding Followers/Following features, Likes, Social sharing, etc. This is a misconception.

ActivityPub / ActivityStreams being Linked Data, allows you to model _any_ business domain, and add federation support to it. Part of my Fediverse advocacy is to highlight this fact so people don't get stuck in a 'just microblogging' mindset.

For this reason I wrote:

- [**De-Emphasize “Being Part of the Fediverse” as primary USP**](https://socialhub.activitypub.rocks/t/1197)
- [**Fediverse Futures: Visions & dreams. Feedback wanted**](https://socialhub.activitypub.rocks/t/1238)

And on the latter subject of [Fediverse Futures](https://socialhub.activitypub.rocks/c/fediversity/fediverse-futures/58) I want to discuss where PubPub may fit in, and maybe entice you to start a new topic of discussion there.

#### Community Has No Boundary

I just found PubPub today, and it seems to have great completeness of vision. Very exciting innovative project looks like! (Though I am less sold on the dependency on Firebase, tbh, but that's off-topic).

PubPub has a Community concept and Collaboration / Social features. What I am trying to get (project / product) developers to think about with Fediverse Futures is considering:

> **What opportunities can Federation bring to your business domain, when incorporated from an early start?**

Sooo many applications are based on a Community concept. Could this be federated? Between PubPub instances? Universally? Why would a community be bound to a single server or cloud-based application? There is a Linked Data draft standards to define community, called [SIOC](http://rdfs.org/sioc/spec/) - Semantically Interlinked Online Communities - that wants to address the problem of _"online community sites [that] are like islands without bridges connecting them"_. This is their proposed ontology:

![SIOC Core Ontology](http://rdfs.org/sioc/spec/img/main_classes_properties.png)

Such an ontology definition can go hand-in-hand with ActivityPub, adding federation support at the business domain level.

In real society communities are part of a complex social fabric. There are many cross-sections and overlaps, they form and are disbanded, they cooperate and unite. They exchange knowledge and information. People are part of many communities at the same time. On the web things change, and the communities are suddenly server-bound, and you need to create accounts with each and every one of them. Many communities are indeed like islands, unaware of other communities, becoming filter-bubbled.

I particularly have this problem of 'community islands' with the [Discourse](https://www.discourse.org/) forum software, a leader in the field. I am juggling way too many forum accounts, jumping in and out, and duplicate content across different threads. To the Discourse community I asked the same question as I will ask of PubPub, and it provides more background to this discussion:

- [**Community has no boundary: Discourse-as-a-Fabric**](https://meta.discourse.org/t/174578)

#### PubPub Federation brainstorm

I would like to know if there's interest in evaluating what Federation support can mean for PubPub, and hopefully this post will inspire you to brainstorm on what is possible when going this route.

If indeed you are interested, and in this thread we can brainstorm a good use case, then I would gladly invite you to post it to SocialHub in a new [Fediverse Futures](https://socialhub.activitypub.rocks/c/fediversity/fediverse-futures/58) topic (you can login with your Github account).

Community has no boundary,
Knowledge is universal,
Brainstorm onwards!

:smiley:

**Comment by gobengo** [comment]
Great idea

**Comment by almereyda** [comment]
While we must generally support such an effort, it is to be taken into account, that before that PubPub would have to render itself completely Free Software, to allow for self-hosting of all dependencies, and as such independent provisioning of its features. Only then an idea of federation starts to make sense, else parts of the application's data will be commited to commercial repositories.

- https://github.com/pubpub/pubpub/issues/1125#issuecomment-723329876 2020-11-06: comment on growth focus
- https://github.com/pubpub/pubpub/discussions/1470#discussioncomment-846095 2021-06-09: considering Supabase as a Firebase replacement

**Reply by aschrijver** [reply]
Very good that you bring that to attention, as I had overlooked this. I especially find the dependence on Firebase problematic, and it would render this solution as unsuitable for me. With Firebase embedded an application is basically part of Google's world-wide surveillance capitalism operation, and makes it no good fit in any initiative where FOSS values and freedoms, as well as privacy are important.

**Reply by gabestein** [reply]
Hey folks — rather than reply in part, I want to take some time to give context on our plan to remove proprietary dependencies and offer better support for self-hosting that would allow for federation. It won't be easy, but we're committed to getting there, and it's baked into our roadmapping and sustainability models. https://notes.knowledgefutures.org/pub/kkcevu38/release/3

**Reply by gobengo** [reply]
@gabestein Thanks for the update. Great plan!

**Reply by aschrijver** [reply]
I agree, it is a great plan, and I really appreciate follow-up and the attention you give to addressing the issues we brought up.

**Comment by metasj** [comment]
Dropping in to comment that I've heard this come up a few times recently in discussion: 
* dweb communities thinking of their community itself as federated (not necessarily willing or able to ask them to all use a central space to publish)
* European communities that want greater local control of data, including who edits when 
* funders and editors of a stream of work, who are considering consolidating their outputs onto their own platform (often: as continuously-updated drafts, which may not be either preprints or final publications; but with the option of expanding to being their own preprint service by adding minimal review, style, and indexing)
